OpenEdge Startup Procedure

mscheblein opened the pull request:
This introduces a new optional Parameter in the Config File (oeStartupProcedure), allowing the User to executed his own code before check syntax, debug or run is executed. I use this to create Database aliases and instantiate singleton classes.

example:
.openedge.json
{ "oeStartupProcedure" : "${workspaceFolder}/set-user.p" }

set-user.p:
/* set the user */
DEF VAR c_imas AS Wawi.Imas .
c_conf = System.Configuration:Instanz .
c_conf:set_user("developer") .
/* do whatever you want */

mscheblein committed:
User defined OE Startup Procedure

mscheblein committed:
Updatet REAME for new Parameter

New Config File Option: oeStartupProcedure

mscheblein committed:
Fix Title

chriscamicas commented on src/shared/openEdgeConfigFile.ts:
```diff
@@ -27,6 +27,7 @@ export interface OpenEdgeConfig {
     parameterFiles?: string[];
     workingDirectory?: string;
     test?: TestConfig;
+    oeStartupProcedure?: string;
```
I think, you don't need to repeat OpenEdge in the naming, especially since we are in an explicit OE config file
`startupProcedure?: string;`
would be enough IMHO

chriscamicas commented:
Thanks again for the contribution.
I think you forgot to update the schema in `openedge.schema.json` 
This file describes the openedge config file JSON schema, and contributes to the VSCode completion when editing it.

mscheblein committed:
added new ConfigPara to Schema

mscheblein commented:
I forgot that this is a thing, should be fixed now

chriscamicas commented:
Thanks, what about the naming ?

mscheblein commented:
Sorry, I do not understand the Question/Problem. Please explain in more Detail.

chriscamicas commented:
In the code review, I suggested renaming `oeStartupProcedure` to `startupProcedure`
since we already are in a `oe` config file, it's explicit enough

mscheblein commented:
> In the code review, I suggested renaming `oeStartupProcedure` to `startupProcedure`
> since we already are in a `oe` config file, it's explicit enough

It says 'no reviews' for me, where would i find your code Review?

chriscamicas commented:
Oh my, I'm so sorry it was "Pending", I guess I missed the Submit button
I just submitted it, sorry

mscheblein committed:
Renamed Startup Parameter

was oeStartupProcedure, is now startupProcedure.
Config File is already OE specific, so the Parameter wont need to start
with "oe"

mscheblein commented:
Guess you are right, it does not need the "OE" in the Name, shortened Parameter Name to "startupProcedure".
